### DISCUSSION TITLE
Supplying configuration via app.config

### DISCUSSION THREAD

**Discussion by huw-dhcw** [cake-build/cake#4113] [Q&A]
I'm trying to call a third party dll (.NET framework 4) which needs configuration to be present in app.config.
Is it possible to arrange this to be present using Cake?

**Comment by pascalberger** [comment]
Can you explain a little bit more detailed what you want to achieve?

Generally Cake is a build orchestrator, not limited to any specific technology or tooling, and you can do everything you can do in any .NET application. For some tools it comes with out of the box convenience routines to call them, for others 3rd party addins provide this, and then there are general purpose aliases to run processes.

**Reply by huw-dhcw** [reply]
"you can do everything you can do in any .NET application" - that's what I'm trying to do. I actually want to use Entity Framework to update a database and the Data provider expects config to be in app.config. Usually this would be called \<appname\>.exe.config, but what would \<appname\> be when running from Cake?
I already have a piece of C# to do what I want which works in LinqPad but only because LinqPad allows me to specify the App.Config entries to use with the query.
